sql/executor.go

```diff
@@ -83,6 +88,7 @@ func (e Executor) Execute(args driver.Request) (driver.Response, int, error) {
 	if planMaker.txn != nil {
 		planMaker.session.Txn = &planMaker.txn.Proto
 		planMaker.session.MutatesSystemDB = planMaker.txn.SystemDBTrigger()
+		planMaker.session.Timestamp = driver.Timestamp(txnTimestamp)
```

Comment:
I think we need to put this timestamp on the `Txn` (which might mean we need a new proto message that wraps `proto.Transaction`). it's really not session state, and this is fragile as a result; if the transaction is committed all its associated state should be blown away.